xml/System/String.xml

```diff
@@ -5287,7 +5287,7 @@ This method performs an ordinal (case-sensitive and culture-insensitive) compari
         <exception cref="T:System.ArgumentNullException">
           <paramref name="value" /> is <see langword="null" />.</exception>
         <block subset="none" type="usage">
-          <para>As explained in [Best Practices for Using Strings](/dotnet/standard/base-types/best-practices-strings), we recommend that you avoid calling string comparison methods that substitute default values and instead call methods that require parameters to be explicitly specified. To determine whether a string ends with a particular substring by using the string comparison rules of the current culture, signal your intention explicitly by calling the <see cref="M:System.String.EndsWith(System.String,System.StringComparison)" /> method overload with a value of <see cref="F:System.StringComparison.CurrentCulture" /> for its <paramref name="comparisonType" /> parameter.</para>
+          <para>As explained in [Best Practices for Using Strings](/dotnet/standard/base-types/best-practices-strings), we recommend that you avoid calling string comparison methods that substitute default values and instead call methods that require parameters to be explicitly specified. To determine whether a string ends with a particular substring by using the string comparison rules of the current culture, signal your intention explicitly by calling the <see cref="M:System.String.EndsWith(System.String,System.StringComparison)" /> method overload with a value of <see cref="F:System.StringComparison.CurrentCulture" /> for its <paramref name="comparisonType" /> parameter. Note that <see cref="F:System.StringComparison.InvariantCulture" /> is rarely appropriate.</para>

```

Comment:
>Note that <see cref="F:System.StringComparison.InvariantCulture" /> is rarely appropriate. [](http://example.com/codeflow?start=683&length=90)

I don't think this is accurate. It is even more confusing. I can see many scenarios that using invariant culture can be desirable. What is the reason adding this note?

Reply:
I just saw @MSDN-WhiteKnight comment and fully agree with it. If I am a user even not familiar with the API and reading this suggested note, for sure I will not get anything from it more than confusion.

Reply:
I have claimed all along that confusion is actually the correct outcome! The API *is*, objectively, confusing, which is why https://github.com/dotnet/designs/pull/207 exists. The *correct* sentiment to instil in the reader of this documentation is "I do not understand this function, so I should read the much more extensive linked documentation about how to use these functions".

> I don't think this is accurate.

I quoted _Best Practices_ which certainly *states* that "On balance, the invariant culture has few properties that make it useful for comparison… One of the few reasons to use [StringComparison.InvariantCulture](https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparison#system-stringcomparison-invariantculture) for comparison is to persist ordered data for a cross-culturally identical display.". I guess you disagree with that too?

Reply:
> I have claimed all along that confusion is actually the correct outcome! The API is, objectively, confusing, which is why https://github.com/dotnet/designs/pull/207 exists. The correct sentiment to instil in the reader of this documentation is "I do not understand this function, so I should read the much more extensive linked documentation about how to use these functions".

Frankly, your added note does not really help at all with what you claim. 

> I quoted Best Practices which certainly states that "On balance, the invariant culture has few properties that make it useful for comparison… One of the few reasons to use [StringComparison.InvariantCulture](https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparison#system-stringcomparison-invariantculture) for comparison is to persist ordered data for a cross-culturally identical display."

This doesn't mean Invariant is rarely used though.

Reply:
Well, I did try months ago to add a note saying "you should not use these functions without reading _Best Practices_", but that was rejected ;)

It's true that I have knowledge only of a small cross-section of .NET's usage; I shall defer to you on the question of whether it is widely used correctly. I've personally literally never (not once!) seen a correct use of `InvariantCulture` in a string comparison method, that's all.